plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestEsriTable.java

```diff
@@ -0,0 +1,127 @@
+            throws IOException
+    {
+        URL resourceLocation = Resources.getResource("esri/counties.json");
+        TrinoFileSystem fileSystem = getConnectorService(getQueryRunner(), TrinoFileSystemFactory.class).create(ConnectorIdentity.ofUser("test"));
+
+        // Create a temporary directory for the table data
+        Location tempDir = Location.of("local:///temp_" + UUID.randomUUID());
+        fileSystem.createDirectory(tempDir);
+        Location dataFile = tempDir.appendPath("counties.json");
+
+        try (OutputStream out = fileSystem.newOutputFile(dataFile).create()) {
+            Resources.copy(resourceLocation, out);
+        }
+
+        List<MaterializedRow> expected = readExpectedResults("esri/counties_expected.txt");
+
+        // ESRI format is read-only, so create data files using the text file format
+        @Language("SQL") String createCountiesTableSql =
+                """
+                CREATE TABLE counties (
+                    name varchar, boundaryshape varbinary)
+                WITH (
+                    format = 'esri',
+                    external_location = '%s')
+                """.formatted(dataFile.parentDirectory());
+        assertUpdate(createCountiesTableSql);
+
+        MaterializedResult result = computeActual("SELECT * FROM counties");
+
+        assertEqualsIgnoreOrder(result.getMaterializedRows(), expected);
+
+        assertQueryFails(
+                "INSERT INTO counties VALUES ('esri fails writes', X'0102030405')",
+                "Writing not supported for StorageFormat\\{serde=com\\.esri\\.hadoop\\.hive\\.serde\\.EsriJsonSerDe, inputFormat=com\\.esri\\.json\\.hadoop\\.EnclosedEsriJsonInputFormat, outputFormat=org\\.apache\\.hadoop\\.hive\\.ql\\.io\\.HiveIgnoreKeyTextOutputFormat\\}"
+        );
+
+        assertUpdate("DROP TABLE counties");
+    }
+
+
+    private static List<MaterializedRow> readExpectedResults(String resourcePath)
+            throws IOException
+    {
+        URL resourceUrl = Resources.getResource(resourcePath);
+        List<String> lines = Resources.readLines(resourceUrl, UTF_8);
+
+        return lines.stream()
+                .map(line -> {
+                    String[] parts = line.split("\t");  // Assuming tab-separated values
+                    return new MaterializedRow(Arrays.asList(
+                            parts[0],  // name
+                            hexToBytes(parts[1])  // hex string for boundaryshape
+                    ));
+                })
+                .collect(toImmutableList());
+    }
+
+    private static byte[] hexToBytes(String hex) {
+        // Remove 'X' prefix, spaces, and single quotes if present
+        hex = hex.replaceAll("^X'|'$", "")  // Remove X' and trailing '
+                .replaceAll("\\s+", "")       // Remove all whitespace
+                .replaceAll("\\*+", "");      // Remove asterisks
+
+        int len = hex.length();
+        byte[] data = new byte[len / 2];
```

Comment:
I think this can use `java.util.HexFormat` and be simplified into something like:
```
return HexFormat.of().parseHex(hex);
```